core/src/main/java/io/grpc/internal/MessageFramer.java

```diff
@@ -342,7 +340,7 @@ private BufferChainOutputStream() {
 
     @Override
     public void write(int b) throws IOException {
-      singleByte[0] = (byte) b;
+      byte[] singleByte = new byte[]{(byte)b};
```

Comment:
Talking about traversing an extra reference during GC is a level of performance discussion that I think we are pretty far away from being impactful. I'm concerned about allocating 1 byte array (which is at least 28 bytes) for every one byte of message; that's something beyond just slow. Given the object is passed to other interfaces, it doesn't seem safe to _assume_ the JVM will stack allocate.